internal/server/users.go

```diff
@@ -107,3 +107,27 @@ func (a *API) UpdateUser(c *gin.Context, r *api.UpdateUserRequest) (*api.User, e
 func (a *API) DeleteUser(c *gin.Context, r *api.Resource) (*api.EmptyResponse, error) {
 	return nil, access.DeleteIdentity(c, r.ID)
 }
+
+func (a *API) DeleteUser2(c *gin.Context, r *api.Resource) (*api.EmptyResponse, error) {
+	if err := access.CanDeleteIdentity(c, r.ID); err != nil {
+		return nil, err
+	}
+
+	if err := access.DeleteAccessKeysForUser(c, r.ID); err != nil {
+		return nil, err
+	}
+
+	if err := access.RemoveUserFromAllGroups(c, r.ID); err != nil {
+		return nil, err
+	}
+
+	if err := access.DeleteCredentialForUser(c, r.ID); err != nil {
+		return nil, err
+	}
+
+	if err := access.DeleteGrantsForUser(c, r.ID); err != nil {
```

Comment:
You can unify all this functionality in one `access.DeleteIdentity` function, it will be simpler to read and follow since the permissions required are the same for all. That function can live in the `internal/server/access/identity.go` file and look something like this:

```
func DeleteUser(c *gin.Context, userID uid.ID) error {
 // check infra admin role
 // check CanDeleteIdentity if not admin
 // delete the access keys for the user (directly in data)
 // remove the user from the groups  (directly in data)
 // delete the credential for the user  (directly in data)
 // delete the grants for the user  (directly in data)
}
```